pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -730,64 +729,72 @@ func (b *MasqueradeBindMechanism) discoverPodNetworkInterface() error {
 		return fmt.Errorf("MTU value out of range ")
 	}
 
-	// Get interface MTU
-	b.dhcpConfig.Mtu = uint16(b.podNicLink.Attrs().MTU)
-
-	err = configureDhcpConfigV4Addresses(b, err)
+	vmIPv4Addr, gatewayIPv4, err := b.parseConfiguredIPv4CIDR()
 	if err != nil {
 		return err
 	}
 
+	b.ipv4Addr = *vmIPv4Addr
+	b.gatewayAddr = gatewayIPv4
+
 	ipv6Enabled, err := b.handler.IsIpv6Enabled(b.podInterfaceName)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to verify whether ipv6 is configured on %s", b.podInterfaceName)
 		return err
 	}
 	if ipv6Enabled {
-		err = configureDhcpConfigV6Addresses(b, err)
+		vmIPv6Addr, gatewayIPv6, err := b.parseConfiguredIPv6CIDR()
 		if err != nil {
 			return err
 		}
+		b.ipv6Addr = *vmIPv6Addr
+		b.gatewayIpv6Addr = gatewayIPv6
 	}
+
+	b.dhcpConfig = b.generateDhcpConfig()
 	return nil
 }
 
-func configureDhcpConfigV4Addresses(b *MasqueradeBindMechanism, err error) error {
-	if b.vmNetworkCIDR == "" {
-		b.vmNetworkCIDR = api.DefaultVMCIDR
+func (b *MasqueradeBindMechanism) generateDhcpConfig() *cache.DhcpConfig {
+	dhcpConfig := &cache.DhcpConfig{
+		Name: b.podInterfaceName,
+		IP:   b.ipv4Addr,
+		IPv6: b.ipv6Addr,
 	}
-
-	gatewayAddr, vmAddr, err := b.generateGatewayAndVmIPAddrs(b.vmNetworkCIDR)
-	if err != nil {
-		return err
+	if b.mac != nil {
+		dhcpConfig.MAC = *b.mac
+	} else {
+		dhcpConfig.MAC = b.podNicLink.Attrs().HardwareAddr
+	}
+	if b.podNicLink != nil {
+		dhcpConfig.Mtu = uint16(b.podNicLink.Attrs().MTU)
+	}
+	if b.gatewayAddr != nil {
+		dhcpConfig.Gateway = b.gatewayAddr.IP.To4()
+	}
+	if b.gatewayIpv6Addr != nil {
+		dhcpConfig.GatewayIpv6 = b.gatewayIpv6Addr.IP.To16()
 	}
 
-	b.dhcpConfig.Gateway = gatewayAddr.IP.To4()
-	b.dhcpConfig.IP = *vmAddr
-	b.gatewayAddr = gatewayAddr
-	b.ipv4Addr = *vmAddr
-	return nil
+	return dhcpConfig
 }
 
-func configureDhcpConfigV6Addresses(b *MasqueradeBindMechanism, err error) error {
-	if b.vmIPv6NetworkCIDR == "" {
-		b.vmIPv6NetworkCIDR = api.DefaultVMIpv6CIDR
+func (b *MasqueradeBindMechanism) parseConfiguredIPv4CIDR() (*netlink.Addr, *netlink.Addr, error) {
+	if b.vmNetworkCIDR == "" {
```

Comment:
Why do we need both `parseConfiguredIPv4CIDR` and `generateGatewayAndVmIPAddrs` can we move this `if` to `generateGatewayAndVmIPAddrs(ipFamily)`